File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        count = {}
        max_app = 0
        result = 0

        for num in nums:
            if num in count:
                count[num] += 1
            else:
                count[num] = 1

            if max_app < count[num]:
                max_app = count[num]
                result = num

        return result

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int el = nums[0], count = 1;

        for (int i = 1; i < nums.size(); i++){
            if (nums[i] == el){
                count++;
            }else{
                count--;

                if (count == 0){
                    el = nums[i];
                    count = 1;
                }
            }
        }

        return el;
    }
};